src/stac_task/models.py

```diff
@@ -0,0 +1,219 @@
+"""Basic models, with little to no logic.
+
+If models are more complicated, they should go in their own module (aka file).
+"""
+
+from __future__ import annotations
+
+from typing import Any, Dict, List, Literal, Optional, Union
+
+import pystac
+import pystac.utils
+from pydantic import BaseModel, ConfigDict
+
+
+class Anything(BaseModel):
+    """A model for any dictionary."""
+
+    model_config = ConfigDict(extra="allow")
+
+
+class UploadOptions(BaseModel):
+    """Options for uploading items and assets after payload execution."""
+
+    path_template: str = r"${collection}"
+    """A string template for specifying the location of uploaded assets.
+    
+    See
+    https://pystac.readthedocs.io/en/stable/api/layout.html#pystac.layout.LayoutTemplate
+    for the available fields.
+    """
+
+    public_assets: Optional[Union[Literal["ALL"], List[str]]] = None
+    """A list of asset keys that should be marked as public when uploaded"""
+
+    headers: Optional[Dict[str, str]] = None
+    """A set of key, value headers to send when uploading data to s3"""
+
+    collections: Optional[Dict[str, str]] = None
+    """A mapping of output collection name to a JSONPath pattern (for matching Items)"""
+
+    s3_urls: bool = False
```

Comment:
Also specific to AWS, but not sure how best to specify something like this.  I think it depends on how stac-asset could eventually support uploads.

Reply:
Yeah, the upload story in **stac-asset** is TBD and will likely mature along with this rewrite, so I'm inclined to keep this for now and then tweak it later once we figure out what **stac-asset** wants.